### DISCUSSION TITLE
Custom controlled relationship field

### DISCUSSION THREAD

**Discussion by Kikky** [payloadcms/payload#1944] [Q&A]
Hello! 

I want to create controlled relationship field with buttons to fill in the data quickly. As I understand I need to use useField() hook and etc. 

But...I can't render this code:
```
{
      name: "value",
      type: "relationship",
      relationTo: "countries",
      hasMany: true,
      admin: {
        components: {
          Field: (props) => {
            return (
              /* here will be buttons to quickly control field's value */
              <RelationshipField
                name={props.name}
                relationTo={props.relationTo}
                hasMany={props.hasMany}
              />
            );
          },
        },
      },
    }
```
Just getting this error ("**No suitable field was found for "Value"**"): 
![image](https://user-images.githubusercontent.com/614922/214333619-43f8a581-42f5-4e0a-b09f-cc8064051321.png)

If I change field type to UI:
```
{
      name: "value",
      type: "ui",
      admin: {
        components: {
          Field: () => {
            return (
              /* here will be buttons to quickly control field's value */
              <RelationshipField name="value" relationTo="countries" hasMany />
            );
          },
        },
      },
    }
```

I can properly render it without errors, but ofc the value isn't stored at DB after update:
![image](https://user-images.githubusercontent.com/614922/214334387-482e4cf5-73db-4e5f-8af1-2b550f2d101c.png)

Can you please help me with this problem? Thanks in advance!

**Comment by jacobsfletch** [comment]
@Kikky what you need to do here is set the field type to `text` and render your custom relationship field as a custom component, similar to your second example but like this:

```ts
{
  name: "value",
  type: "text",
  admin: {
    components: {
      Field: () => {
         // render anything you want here, including the Payload relationship component
         // interact with form state as needed using form hooks
         // then, save your relationship ID as a string to this text field using field hooks
      }
    },
  },
}
```

Give this a shot and let me know how it goes 👍

**Reply by Kikky** [reply]
> ```ts
> / then, save your relationship ID as a
> ```

Thanks for your reply! But I have hasMany={true} on this field and this means that I have to store its value (an array of ids) as a string and parse it after reading from DB...and also I can't get related fields via GraphQL.

I want to keep data format in DB and advantages of relationship field, but to add some UI with functuionality.
